script/test.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+SIMPLEFS_MOD=simplefs.ko
+IMAGE=$1
+IMAGESIZE=$2
+MKFS=$3
+
+D_MOD="drwxr-xr-x"
+F_MOD="-rw-r--r--"
+S_MOD="lrwxrwxrwx"
+MAXFILESIZE=4194304
+
+test_op() {
+    local op=$1 
+    echo
+    echo -n "Testing cmd: $op..."
+    sudo sh -c "$op" >/dev/null && echo "Success"
+}
+
+check_exist() {
+    local mode=$1
+    local nlink=$2
+    local name=$3
+    echo
+    echo -n "Check if exist: $mode $nlink $name..."
+    sudo ls -lR  | grep -e "$mode $nlink".*$name >/dev/null && echo "Success" || \
+    echo "Failed" 
+}
+
+if [ "$EUID" -eq 0 ]
+  then echo "Don't run this script as root"
+  exit
+fi
+
+mkdir -p test  
+sudo umount test 2>/dev/null
+sleep 1
+sudo rmmod $SIMPLEFS_MOD 2>/dev/null
+sleep 1
+(modinfo $SIMPLEFS_MOD || exit 1) && \
+echo && \
+sudo insmod $SIMPLEFS_MOD  && \
+dd if=/dev/zero of=$IMAGE bs=1M count=$IMAGESIZE status=none&& \
+./$MKFS $IMAGE && \
+sudo mount -t simplefs -o loop $IMAGE test && \
+cd test
```

Comment:
You can make use of `pushd` and `popd`.

Reply:
Thanks for review, I've replaced `cd` with `pushd` and `popd`.